src/Bridges/graph.jl

```diff
@@ -0,0 +1,269 @@
+    # objective node index -> Index of bridge to be used
+    objective_best::Vector{Int}
+    objective_last_correct::Int
+end
+function Graph()
+    return Graph(
+        Vector{Edge}[], ConstraintNode[], Int[], Int[], Int[], 0,
+        Vector{Edge}[], Int[], Int[], 0,
+        Vector{ObjectiveEdge}[], Int[], Int[], 0,
+    )
+end
+
+function Base.show(io::IO, graph::Graph)
+    print(io, "Bridge graph with ")
+    print(io, length(graph.variable_best), " variable nodes, ")
+    print(io, length(graph.constraint_best), " variable nodes and ")
+    print(io, length(graph.objective_best), " objective nodes.")
+end
+function variable_nodes(graph::Graph)
+    return MOIU.LazyMap{VariableNode}(
+        i -> VariableNode(i), eachindex(graph.variable_best))
+end
+function constraint_nodes(graph::Graph)
+    return MOIU.LazyMap{ConstraintNode}(
+        i -> ConstraintNode(i), eachindex(graph.constraint_best)
+    )
+end
+function objective_nodes(graph::Graph)
+    return MOIU.LazyMap{ObjectiveNode}(
+        i -> ObjectiveNode(i), eachindex(graph.objective_best))
+end
+
+# After `add_bridge(b, BT)`, some constrained variables `(S,)` in
+# `keys(b.variable_best)` or constraints `(F, S)` in `keys(b.constraint_best)`
+# or `(F,)` in `keys(b.objective_best)` may be bridged
+# with less bridges than `b.variable_dist[(S,)]`,
+# `b.constraint_dist[(F, S)]` or `b.objective_dist[(F,)]` using `BT`.
+# We could either recompute the distance from every node or clear the
+# dictionary so that the distance is computed lazily at the next `supports_constraint`
+# call. We prefer clearing the dictionaries so as this is called for each
+# bridge added and recomputing the distance for each bridge would be a wase
+# if several bridges are added consecutively.
+function Base.empty!(graph::Graph)
+    empty!(graph.variable_edges)
+    empty!(graph.variable_constraint_node)
+    empty!(graph.variable_constraint_cost)
+    empty!(graph.variable_dist)
+    empty!(graph.variable_best)
+    graph.variable_last_correct = 0
+    empty!(graph.constraint_edges)
+    empty!(graph.constraint_dist)
+    empty!(graph.constraint_best)
+    graph.constraint_last_correct = 0
+    empty!(graph.objective_edges)
+    empty!(graph.objective_dist)
+    empty!(graph.objective_best)
+    graph.objective_last_correct = 0
+end
+
+function add_edge(graph::Graph, node::VariableNode, edge::Edge)
+    push!(graph.variable_edges[node.index], edge)
+end
+function add_edge(graph::Graph, node::ConstraintNode, edge::Edge)
+    push!(graph.constraint_edges[node.index], edge)
+end
+function add_edge(graph::Graph, node::ObjectiveNode, edge::ObjectiveEdge)
+    push!(graph.objective_edges[node.index], edge)
+end
+function add_variable_node(graph::Graph)
+    push!(graph.variable_edges, Edge[])
+    # Use an invalid index so that the code errors instead return something
+    # incorrect in case `set_variable_constraint_node` is not called.
+    push!(graph.variable_constraint_node, ConstraintNode(-1))
```

Comment:
It might be more readable to define a constant instead of using `-1` since this check is used elsewhere.